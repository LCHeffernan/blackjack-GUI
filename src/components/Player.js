import React from "react";
import PropTypes from "prop-types";
import Cards from "./Cards";
import "../styles/player.css";

const Player = ({ playerHand, playerScore }) => (
  <div className="cards-container player" data-testid="player-cards">
    {playerHand && (
      <div className="cards">
        <div className="score">{`Player score: ${playerScore}`}</div>
        <Cards cards={playerHand} />
      </div>
    )}
  </div>
);

Player.defaultProps = {
  playerHand: [],
};

Player.propTypes = {
  playerHand: PropTypes.arrayOf(
    PropTypes.shape({
      cardRank: PropTypes.string,
      cardSuit: PropTypes.string,
      cardValue: PropTypes.number,
    })
  ),
  playerScore: PropTypes.number.isRequired,
};

export default Player;
