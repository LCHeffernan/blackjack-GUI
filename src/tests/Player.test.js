import React, { render, screen } from "@testing-library/react";
import Player from "../components/Player";

describe("Player", () => {
  const validProps = {
    playerHand: [
      {
        cardRank: "A",
        cardSuit: "Mock",
        cardValue: 11,
      },
      {
        cardRank: "K",
        cardSuit: "Mock",
        cardValue: 10,
      },
    ],
    playerScore: 21,
  };
  it("renders correctly", () => {
    const { asFragment } = render(
      <Player
        playerHand={validProps.playerHand}
        playerScore={validProps.playerScore}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders the correct score", () => {
    render(
      <Player
        playerHand={validProps.playerHand}
        playerScore={validProps.playerScore}
      />
    );
    expect(screen.getByText("Player score: 21")).toHaveClass("score");
  });
});
