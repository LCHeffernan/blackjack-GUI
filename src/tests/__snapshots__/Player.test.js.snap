// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Player renders correctly 1`] = `
<DocumentFragment>
  <div
    class="cards-container player"
    data-testid="player-cards"
  >
    <div
      class="cards"
    >
      <div
        class="score"
      >
        Player score: 21
      </div>
      <div
        class="player-hand"
      >
        <div
          class="card-container"
        >
          <div
            class="top-left-rank"
          >
            A
          </div>
          <div
            class="suit"
          >
            Mock
          </div>
          <div
            class="bottom-right-rank"
          >
            A
          </div>
        </div>
        <div
          class="card-container"
        >
          <div
            class="top-left-rank"
          >
            K
          </div>
          <div
            class="suit"
          >
            Mock
          </div>
          <div
            class="bottom-right-rank"
          >
            K
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
